File: Classes/Premade/MASPreferences/PreferenceWindow.m
//
//  PreferenceWindow.m
//  Foodie
//
//  Created by Jon Brown on 3/18/13.
//  Copyright (c) 2013 Jon Brown. All rights reserved.
//

#import "PreferenceWindow.h"
#import "MASPreferencesWindowController.h"
#import "GeneralPreferencesViewController.h"
#import "AdvancedPreferencesViewController.h"
#import "UpdatePreferenceViewController.h"

@implementation PreferenceWindow


#pragma mark -

- (IBAction)openPreferences:(id)sender
{
    
    [self.preferencesWindowController showWindow:nil];
    
}


#pragma mark - Public accessors

- (NSWindowController *)preferencesWindowController
{
    if (_preferencesWindowController == nil)
    {
        NSViewController *generalViewController = [[GeneralPreferencesViewController alloc] init];
        NSViewController *advancedViewController = [[AdvancedPreferencesViewController alloc] init];
        NSViewController *updateViewController = [[UpdatePreferenceViewController alloc] init];
        
        
        
#ifdef TRIAL
        
    NSArray *controllers = [[NSArray alloc] initWithObjects:generalViewController, updateViewController, nil];

#elif WEBSITE
    
    NSArray *controllers = [[NSArray alloc] initWithObjects:generalViewController, updateViewController, nil];
        
#elif STORE
        
    NSArray *controllers = [[NSArray alloc] initWithObjects:generalViewController, nil];

#else
        
    NSArray *controllers = [[NSArray alloc] initWithObjects:generalViewController, nil];
        
        
#endif
        
        
        
        
       // [generalViewController release];
      //  [advancedViewController release];
     //   [updateViewController release];
        
        NSString *title = NSLocalizedString(@"Preferences", @"Common title for Preferences window");
        _preferencesWindowController = [[MASPreferencesWindowController alloc] initWithViewControllers:controllers title:title];
      //  [controllers release];
    }
    return _preferencesWindowController;
}


- (void) awakeFromNib
{
    
    
    [prefwindow setBackgroundColor:[NSColor colorWithPatternImage:[NSImage imageNamed:@"Black"]]];
    
}


@end

File: Classes/Premade/MASPreferences/PreferenceWindow.swift
//
//  PreferenceWindow.swift
//  BeFit
//
//  Created by Andrej Slegl on 11/8/19.
//

import Foundation
import Cocoa
import MASPreferences

class PreferenceWindow: NSObject {
    private lazy var preferencesWindowController: NSWindowController = {
        let generalViewController = GeneralPreferencesViewController()
        let updateViewController = UpdatePreferenceViewController()
        
        #if WEBSITE
        let controllers: [NSViewController] = [generalViewController, updateViewController]
        #else
        let controllers: [NSViewController] = [generalViewController]
        #endif
        
        let title = NSLocalizedString("Preferences", comment: "Common title for Preferences window")
        let vc = MASPreferencesWindowController(viewControllers: controllers, title: title)
        //vc.window?.backgroundColor = NSColor(patternImage: NSImage(named: "Black")!)
        return vc
    }()
    
    @IBAction func openPreferences(_ sender: AnyObject) {
        preferencesWindowController.showWindow(nil)
    }
}
